benchmarks/cudf-merge.py

```diff
@@ -206,10 +206,17 @@ async def worker(rank, eps, args):
         iter_timings = []
 
         iter_t = clock()
-        await distributed_join(args, rank, eps, df1, df2, iter_timings)
+        ret = await distributed_join(args, rank, eps, df1, df2, iter_timings)
         await barrier(rank, eps)
         iter_took = clock() - iter_t
 
+        # Ensure the number of matches falls within `args.frac_match` +/- 1%
+        expected_len = args.chunk_size * args.frac_match
+        expected_len_err = expected_len * 0.01
+        assert len(ret) in range(
+            expected_len - expected_len_err, expected_len + expected_len_err
+        )
```

Comment:
I think that the dataframe generation is such that `allreduce(len(ret), op=SUM)` is exactly equal to `expected_len`, so perhaps do that? A bad hash could result in this assertion failing even if the collective result is correct.

Reply:
In this particular benchmark we don't do `allreduce`, maybe we could introduce this in the future but I'll try not to tackle everything in this one PR. IMO, if we have a bad hash and the assertion fails, this is probably worth causing an outright failure to be investigated and the benchmarks made more robust, what do you think?

Reply:
Yes, I think that's reasonable.